### ISSUE TITLE
New outreach samples and Higgs to four lepton analysis

### ISSUE THREAD

**Issue by stwunsch** [cernopendata/opendata.cern.ch#3110]
Hi all!

I've missed up to now to make the samples of the ROOT tutorial [here](https://root.cern/doc/master/df103__NanoAODHiggsAnalysis_8C.html) freely available. It's performing the Higgs to four lepton analysis based on the NanoAOD-like samples, see the plot below!

![](https://github.com/stwunsch/HiggsToFourLeptonsNanoAODOutreachAnalysis/raw/master/plots/combined_Higgs_mass.png)

I'd like to integrate the analysis (see the repo [here](https://github.com/stwunsch/HiggsToFourLeptonsNanoAODOutreachAnalysis)) and also the samples created based on v1.3 of the AOD2NanoAODOutreachTool in the portal. I've already all available on my disk, just missing sharing it!

Some questions below :)

Best,
Stefan

**Comment by stwunsch** [comment]
I would create a page like https://opendata.cern.ch/record/12350 and link to each of the new samples with dedicated pages like https://opendata.cern.ch/record/12351. The files I've created contain actually all collections, also jets and taus, which are not really needed by the Higgs to four lepton analysis. Should we keep it like this anyway?

**Comment by katilp** [comment]
Excellent, thanks! @caredg how would you see it? We could have this code in a record as Stefan suggests but the samples could eventually be those that will be produced by the IRIS-HEP team. 

We need to do a little bit of thinking in which directory on eospublic we store the samples.
That's not directly exposed to external users but to have them in good order would be helpful for the future.

For the moment, what we have agreed on is to have them under derived-data (reflecting the fact that they are not from an official production) and under the producer name i.e.

```
-bash-4.2$ eos ls -l  /eos/opendata/cms/derived-data
drwxr-xr-+   1 simko    us       118736497680 Sep 30  2020 AOD2NanoAODOutreachTool
drwxr-xr-+   1 simko    us           28787601 Jun 27  2019 run1-physics-candidates
-bash-4.2$ eos ls -l  /eos/opendata/cms/derived-data/AOD2NanoAODOutreachTool/
-rw-r--r--   2 swunsch  zh         9247252629 Aug 13  2019 DYJetsToLL.root
-rw-r--r--   2 swunsch  zh          196517149 Aug 13  2019 GluGluToHToTauTau.root
-rw-r--r--   2 swunsch  zh         2244449133 May 23  2019 Run2012BC_DoubleMuParked_Muons.root
-rw-r--r--   2 swunsch  zh        10918632568 Aug 13  2019 Run2012B_TauPlusX.root
-rw-r--r--   2 swunsch  zh        15886107547 Aug 13  2019 Run2012C_TauPlusX.root
-rw-r--r--   2 swunsch  zh         3528995041 Aug 13  2019 TTbar.root
-rw-r--r--   2 swunsch  zh          231849713 Aug 13  2019 VBF_HToTauTau.root
-rw-r--r--   2 swunsch  zh        10731401995 Aug 13  2019 W1JetsToLNu.root
-rw-r--r--   2 swunsch  zh        12161887620 Aug 13  2019 W2JetsToLNu.root
-rw-r--r--   2 swunsch  zh         6544850738 Aug 13  2019 W3JetsToLNu.root
drwxr-xr-+   1 simko    us        47044553547 Sep 30  2020 ws1.0
```

Note that we added _Muons to the DoubleMu file to indicate that it only contained muons. The rest are those used in the HTauTau example.

As AOD2NanoAODOutreachTool has become quite popular, this could profit from an additional directory layer e.g. with versioning of AOD2NanoAODOutreachTool, or something indicating the event content.  What do you think?

**Comment by stwunsch** [comment]
Also take care that the current "generic" samples like `TTbar.root` don't contain electron and photons! I guess IRIS HEP would also take care of these?

**Comment by stwunsch** [comment]
In case they will update anyway all samples to a generic set, we could also push first my reduced set to get the example running and replace them later with an updated version including more objects.

**Comment by stwunsch** [comment]
@katilp @caredg Ping :)

**Comment by caredg** [comment]
Thanks a lot @stwunsch for putting all this together.  @katilp, after giving this more thought, I would actually proceed as Stefan's original proposal.  I.e., creating a page like https://opendata.cern.ch/record/12350 and link to each of the new samples with dedicated pages like https://opendata.cern.ch/record/12351. In that way, the example is nice and complete and there will be no confusion with event content as the links, such the record 12351 above, contain the description of information present.
I would shorten the repository name a bit as "HiggsTo4LeptonsNanoAODOutreachAnalysis".

Now, regarding the storage, yes I guess it would be wise to separate this versions.  How about directories like:
AOD2NanoAODOutreachTool
        |------------------------------------> ForHiggsToTauTau
        |------------------------------------> ForHiggsTo4Leptons
        |------------------------------------> Other 
        |------------------------------------> IrisHEP 

However, I suppose moving the original files to `ForHiggsToTauTau` would mean making changes in the released code (the directory is hardoced in skim.cxx) and I that could be problematic.  So, if this is a problem, we could simply add the rest of the directories and leave the original HToTauTau at the top of the main directory.

Also, I do no think `Run2012BC_DoubleMuParked_Muons.root` is used in any example described in the portal.  If this is going to be replaced anyway, can we just delete this one? The ws1.0.root was a workshop version that could probably be erased in the near future or replaced.  If we want to keep these spurious ones, however, I would move them `Other`.  In IrisHEP (or Comprenhensive, etc.; whatever name suits), we can put the more complete ones.

what do you think?

**Comment by katilp** [comment]
Thanks, sounds good to me. For the muon file, it is used in a http://opendata.cern.ch/record/12342

**Comment by caredg** [comment]
Right, ok.  Then we should keep that one (`Run2012BC_DoubleMuParked_Muons.root`) in, of course.

**Comment by stwunsch** [comment]
@tiborsimko (or others, if you know) Is it possible to have symlink-like links on EOS? Then we could also keep the current file location valid but move them physically.

**Comment by stwunsch** [comment]
@katilp @caredg Thanks for your response! I'll prepare the records for the new example in the next two weeks :)

**Comment by stwunsch** [comment]
@katilp @caredg The records are now available in the PR #3113. I've also moved the analysis repository to https://github.com/cms-opendata-analyses/HiggsToFourLeptonsNanoAODOutreachAnalysis.

**Comment by tiborsimko** [comment]
> @tiborsimko (or others, if you know) Is it possible to have symlink-like links on EOS? 

Yes, one can use the `eos file symlink` command. I have tested that XRootD exposure recognises the old URIs well too.
